src/Hazelcast.Net.Tests/Remote/ClientMapTest.cs

```diff
@@ -1407,5 +1408,42 @@ public async Task TestValuesPredicate()
             Assert.IsTrue(enumerator.MoveNext());
             Assert.AreEqual("value1", enumerator.Current);
         }
+
+        [Test]
+        public async Task TestTTLUpdated()
+        {
+            var map = await Client.GetMapAsync<string, string>(CreateUniqueName());
+            // in seconds
+            var maxTTL = 100;
+            var minTTL = 3;
+            var key = "key1";
+            var latch = new ManualResetEvent(false);
+            
+            var starTime = Stopwatch.StartNew();
+            await map.SetAsync(key, "value1", TimeSpan.FromSeconds(maxTTL));
+
+            // Notify when key removed
+            var parallelCheck = Task.Run(async () =>
+            {
+                await AssertEx.SucceedsEventually(
+                    async () => Assert.False(await map.ContainsKeyAsync(key)),
+                    maxTTL * 1000,
+                    500);
+                latch.Set();
+            });
+
+            // Reduce the TTL
+            Assert.True(await map.UpdateTimeToLive(key, TimeSpan.FromSeconds(minTTL)));
+
+            await latch.WaitOneAsync();
+            starTime.Stop();
+            var elapsed = starTime.ElapsedMilliseconds;
+
+            // Definitively, less then maxTTL
+            Assert.Less(elapsed, maxTTL * 1000);
+
+            // Close to min TTL
+            Assert.Less(elapsed, minTTL * 1000 * 2);
```

Comment:
i am not sure if this will always work if the server is working slow, etc.
Does the ContainsKeyAsync trigger eviction check on expiry, if so, it may be safe but if we do it on scheduled basis on the server side, there may be a time diff until that scheduler is triggered and entry is evicted. If that is the case, this check may fail depending on the scheduler interval.

Reply:
Right, it may not work well in slow cases. Do you have any suggestion?

Reply:
Removed the assertion at https://github.com/hazelcast/hazelcast-csharp-client/pull/997/commits/147b40ab53b751bef4acdc003c0d22faf87dad87 There is a public api coeverage work upcoming. I don't know the tests but I'll look for more according to results of that work.